DESIGN.md

```diff
@@ -54,39 +56,113 @@ Usage:
 -e --execute ["text"|script-file|STDIN]
 ```
 
-## Meta Commands
+## Command interface
+
+Users can type three types of entries into the shell:
+
+1. [PartiQL statements](https://partiql.org/)., 
+1. Database commands, and
+1. Shell commands.
+
+### PartiQL statements
+These statements are any valid PartiQL syntax. These queries will be captured as is and sent to the server for
+evaluation with the results returned and displayed in the specified format.
+
+### Database commands
+These commands are specifically limited to those specified here and relate to the principle of least astonishment.
+These are the types of commands users will be used to seeing in other database shells. These aren't queries and cannot
+be executed by the database query engine, rather they are commands at the level of the database itself, rather than
+the data in the database.
+
+* `start transaction` or `begin`
+  * This starts a transaction. If we are in auto-commit mode, the shell reports a warning saying that this has no 
+    effect. 
+* `commit`
+  * This commits a transaction. If there is no transaction in progress, the shell reports an error saying that there is 
+    no active transaction. If we are in auto-commit mode, the command is accepted and no error or warning or information
+    is reported.
+* `use [ledger|endpoint] ledger-or-endpoint`
+  * Connect to a different endpoint or ledger.
+* `show [ledgers|tables]`
+  * Display a list of tables in the current ledger, or ledgers at the current endpoint, depending on the command 
+    arguments.
```

Comment:
I'm not sure why these should be in the second level.

Reply:
`use database` is in T-SQL and MySQL, `show tables` is in MySQL. I think these are common enough to be second-level "natural" command.

Reply:
Not going to die on this hill though if you think this is wrong :)

Reply:
`use ledger` is not really the same as `use database` in MySQL. It's called a "database" in MySQL, but it's really just a namespace on the same server (e.g. you can even do transactions across namespaces).

I think `show tables` just shouldn't be a second tier command at all.

I guess I'd flip the question around too - almost nothing should be in the second tier. `help` and `quit` are obvious ones and I think `start ..` and `commit` are too. Especially given that we *could* have chosen to implement `commit` in terms of `tx.execte_statement("commit")`.

Reply:
Cool, will update.